config.ru

```diff
@@ -11,6 +11,19 @@ map '/lobster' do
   run Rack::Lobster.new
 end
 
+map '/headers' do
+  headers = proc do |env|
+    [200, { "Content-Type" => "text/html" }, [
+      env.select {|key,val| key.start_with? 'HTTP_'}
+      .collect {|key, val| [key.sub(/^HTTP_/, ''), val]}
+      .collect {|key, val| CGI::escapeHTML "#{key}: #{val}"}
+      .sort
+      .join("<br/>\n")
+    ]]
+  end
+  run headers
```

Comment:
In Firefox it shows "Quirks mode". HTML is very minimal. Just `<br/>` for new lines and strings are HTML escaped for proper output.

In any case here is an example raw output of current implementation:
```
ACCEPT: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8<br/>
ACCEPT_ENCODING: gzip, deflate<br/>
ACCEPT_LANGUAGE: bg,en-US;q=0.7,en;q=0.3<br/>
COOKIE: s_fid=1C893BDD6A1DFF78-0BE36C33F7C57F7E; s_vi=[CS]v1|2CB1B999851D4D11-60000136A00064DA[AC]; AMCV_945F02BE332957400A490D4C%40AdobeOrg=-<snip for security>4882679239%26vn%3D12<br/>
DNT: 1<br/>
FORWARDED: for=10.3.2.11;host=ruby-ex-aosqe.example.com;proto=http<br/>
HOST: ruby-ex-aosqe..com<br/>
UPGRADE_INSECURE_REQUESTS: 1<br/>
USER_AGENT: Mozilla/5.0 (X11; Fedora; Linux x86_64; rv:54.0) Gecko/20100101 Firefox/54.0<br/>
VERSION: HTTP/1.1<br/>
X_FORWARDED_FOR: 10.3.2.11<br/>
X_FORWARDED_HOST: ruby-ex-aosqe.example.com<br/>
X_FORWARDED_PORT: 80<br/>
X_FORWARDED_PROTO: http
```